src/backend/joanie/signature/backends/lex_persona/client.py

```diff
@@ -0,0 +1,400 @@
+            data = json.loads(request.body)
+            try:
+                webhook_event_id = data["id"]
+                response = self._verify_webhook_event(webhook_event_id)
+            except KeyError as error:
+                logger.error("There is no ID key in the request body :%s", error)
+                return Response(
+                    {"error": "Failed, missing 'id' information in request body."},
+                    status=404,
+                )
+
+        if response is not None and response.status_code == 404:
+            return response
+        if response is not None and response.status_code == 200:
+            return webhooks.synchronize_signed_document(request)
+
+        return Response(
+            {"error": "Unknown error occurred"},
+            status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+        )
+
+    def upload_file(self, *args, **kwargs):
+        """
+        Wrapper to start a signature procedure with 2 signers.
+        This method implies : to prepare a workflow, to upload
+        a document, and start the process to sign them.
+        """
+        self._create_workflow()
+        self._upload_file_to_workflow()
+        self._update_workflowstatus_to_start()
+        return self.workflow_id
+
+    def create_invitation_link(self, recipient_email: str, *args, **kwargs):
+        """
+        Create invitation link to sign the document for 1 email address.
+        It will bring the user to sign every document that are linked to his email address.
+        """
+        return self._create_invitation_link_to_sign(recipient_email=recipient_email)
+
+    def sign_selection_of_document(self, recipient_email: str, workflow_ids: list):
+        """
+        Create the link to go sign directly the selected documents by their workflow id.
+        """
+        token_jwt = self._get_jwt_token_from_invite_url(recipient_email=recipient_email)
+        if not token_jwt:
+            return "Token can not be extracted from the URL invite link"
+
+        response = self._sign_specific_selection_of_workflow(
+            workflow_ids=workflow_ids, token_jwt=token_jwt
+        )
+        return {"inviteUrlSelection": response["consentPageUrl"]}
+
+    def refuse_signature(self, recipient_email: str, workflow_ids: list):
+        """
+        Refuse the invite of a workflow.
+        """
+        token_jwt = self._get_jwt_token_from_invite_url(recipient_email=recipient_email)
+        if not token_jwt:
+            return "Token can not be extracted from the URL invite link"
+
+        for workflow_id in workflow_ids:
+            url = f"{self.base_url}/api/workflows/{workflow_id}/refuse"
+            headers = {
+                "Authorization": f"Bearer {token_jwt}",
+                "Content-Type": "application/json",
+            }
+            payload = {"reason": "I disagree."}
+            response = requests.post(url, headers=headers, json=payload, timeout=3)
+            return response.json()
```

Comment:
This will not loop on all workflow_ids

Reply:
I've updated the for loop the folling way :
```python
        responses = []
        for workflow_id in workflow_ids:
            url = f"{self.base_url}/api/workflows/{workflow_id}/refuse"
            headers = {
                "Authorization": f"Bearer {token_jwt}",
                "Content-Type": "application/json",
            }
            payload = {"reason": "I disagree."}
            response = requests.post(url, headers=headers, json=payload, timeout=3)
            responses.append(response)
        return Response({"responses": responses}, status=status.HTTP_200_OK)
```